web-app/src/app/screens/Feeds/index.tsx

```diff
@@ -71,11 +71,11 @@ export default function Feed(): React.ReactElement {
   const [selectedFeedTypes, setSelectedFeedTypes] = useState(
     getInitialSelectedFeedTypes(searchParams),
   );
-  const [activeSearch, setActiveSearch] = useState('');
+  const [activeSearch, setActiveSearch] = useState(searchParams.get('q') ?? '');
+  const [searchQuery, setSearchQuery] = useState(activeSearch);
```

Comment:
[suggestion]: Having two state variables with the same "meaning" can be confusing and make it hard to follow the code. If possible, I suggest using a single state variable.

Reply:
I intentionally use a 'duplicate' searchQuery to address the original issue. In line 178, the onChange function only updates the searchQuery, not the activeSearch, which is responsible for triggering an actual search. That's why it removed a keystroke when the back navigation button is clicked.